### PR TITLE
roots now give correct result 

### LINKED ISSUE #16589: roots gives incorrect result
`roots(x**4 - 8*sqrt(2)*x**3 + 4*x**3 - 64*sqrt(2)*x**2 + 1024*x, x)` gives a set of roots, among which there is no `x=0` and there is `x=8sqrt(2)/3`, but it is not a root.

### PR DESCRIPTION
**Pull request by vorabrijesh**
Fixes #16589
#### Brief description of what is fixed or changed
`roots(x**4 - 8*sqrt(2)*x**3 + 4*x**3 - 64*sqrt(2)*x**2 + 1024*x)`
now gives correct result
#### Other comments
although` roots(x**4 - 8*sqrt(2)*x**3 + 4*x**3 - 64*sqrt(2)*x**2 + 1024*x, x)` still doesn't give correct ans

#### Release Notes

* polys
  * added solve() to solve multivariate polynomial

### CODE REVIEW

File: sympy/polys/polyroots.py [hunk 1]
```diff
@@ -845,6 +845,7 @@ def roots(f, *gens, **flags):
 
     """
     from sympy.polys.polytools import to_rational_coeffs
+    from sympy.solvers.solvers import solve
```
Comment (smichr) [inline sympy/polys/polyroots.py]: `solve` is a high-level routine that should not be called from this low level routine. The diff I gave (and now slightly momdified) will handle the case when a generator is not specified.

**Comment by smichr** [comment]
:exclamation:  Hmm...Travis is showing that this passed but if you click through via Details you will see that it is picking up the run for the 'case of variables in autolev parsing' which is #16580 .

ping @asmeurer , @oscargus , @oscarbenjamin

**Comment by oscarbenjamin** [comment]
No idea. Maybe it's a bug to do with build cancellation some how. Looks like the build for this PR got cancelled:
[https://travis-ci.org/sympy/sympy/builds/517872868](https://travis-ci.org/sympy/sympy/builds/517872868)

**Comment by Sc0rpi0n101** [comment]
The commit in this PR is also listed as the only commit in #16580 and the diff is also the same.
@vorabrijesh  What did you do?

**Comment by vorabrijesh** [comment]
Actually, I pushed the commit in the wrong branch

**Comment by smichr** [comment]
I am closing this since there is some doubt about the run and the fix should not involve using solve here.

**Pull request closed without merging**